### ISSUE TITLE
is there anywhere a collection of tests for the input grammar?

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo-phi-normalizer#562]
I'm not sure whether this grammar is valid: https://github.com/objectionary/eo-phi-normalizer/blob/master/eo-phi-normalizer/grammar/EO/Phi/Syntax.cf How can I make sure it is? Do you have a list of test files, both for positive testing (they all should be parseable) and negative (they all should lead to parsing errors)?

**Comment by yegor256** [comment]
@deemp please, check

**Comment by yegor256** [comment]
@deemp of course, [here](https://github.com/objectionary/eo/tree/master/eo-maven-plugin/src/test/resources/org/eolang/maven/phi), and [here](https://github.com/objectionary/eo/tree/master/eo-maven-plugin/src/test/resources/org/eolang/maven/unphi), and [here](https://github.com/objectionary/eo/tree/master/eo-parser/src/test/resources/org/eolang/parser/packs/syntax) (for EO syntax). It's helpful for catching bugs earlier, at the level of syntax. Simply adding a file to any of these directories automatically puts it into the testing pipeline.

I've also added [positive](https://github.com/objectionary/eo/tree/master/eo-parser/src/test/resources/org/eolang/parser/phi-syntax) and [negative](https://github.com/objectionary/eo/tree/master/eo-parser/src/test/resources/org/eolang/parser/phi-typos) test packs for phi (maybe you can reuse them).

**Comment by deemp** [comment]
@yegor256, the positive and negative tests for phi pass now.

Parser tests are read and executed here:
https://github.com/objectionary/eo-phi-normalizer/blob/b7926285aa1495b07806f67fa4d05dd6c437696e/eo-phi-normalizer/test/Language/EO/Phi/ParserSpec.hs

They're defined here:
https://github.com/objectionary/eo-phi-normalizer/blob/b7926285aa1495b07806f67fa4d05dd6c437696e/eo-phi-normalizer/test/eo/phi/parser/expressions.yaml

**Comment by yegor256** [comment]
@deemp perfect!
